### PR TITLE
Add new device types

### PR DESCRIPTION
**Pull request by iMicknl**
- Added support for AirSensor devices
- Added support for ElectricitySensor devices
- Added support for Curtain devices
- Added support for Generic devices (cover)
- Added support for SwingingShutter devices

Based on https://github.com/dubocr/homebridge-tahoma/blob/master/mapping.json

### CODE REVIEW

File: custom_components/tahoma/sensor.py [hunk 1]
```diff
@@ -55,23 +57,39 @@ def unit_of_measurement(self):
...
+            return TEMP_CELSIUS
 
         if self.tahoma_device.uiclass == "HumiditySensor":
             return UNIT_PERCENTAGE
 
         if self.tahoma_device.uiclass == "LightSensor":
             return "lx"
 
+        if self.tahoma_device.uiclass == "ElectricitySensor":
+            return "W"
```
Comment (vlebourl) [inline custom_components/tahoma/sensor.py:70]: Is it W or Wh or kWh?

Comment (vlebourl) [reply custom_components/tahoma/sensor.py:70]: if the former, we should use `homeassistant.const.POWER_WATT`, 
if one of the laters, we should use `homeassistant.const.ENERGY_WATT_HOUR` or `homeassistant.const.ENERGY_KILO_WATT_HOUR`

File: custom_components/tahoma/sensor.py [hunk 2]
```diff
@@ -55,23 +57,39 @@ def unit_of_measurement(self):
...
+
+        if self.tahoma_device.uiclass == "AirSensor":
+            return "ppm"
```
Comment (vlebourl) [inline custom_components/tahoma/sensor.py:73]: we should use `homeassistant.const.CONCENTRATION_PARTS_PER_MILLION`

File: custom_components/tahoma/sensor.py [hunk 3]
```diff
@@ -86,15 +104,29 @@ def update(self):
         if CORE_RELATIVE_HUMIDITY_STATE in self.tahoma_device.active_states:
             self.current_value = float(
                 "{:.2f}".format(
```
Comment (vlebourl) [inline custom_components/tahoma/sensor.py:106]: I had a review on a PR in core asking to use f-strings. should we use them instead of `"".format`?

Comment (vlebourl) [reply custom_components/tahoma/sensor.py:106]: `f"{self.tahoma_device.active_states.get(CORE_RELATIVE_HUMIDITY_STATE):.2f}"`

Comment (iMicknl) [reply custom_components/tahoma/sensor.py:106]: Not sure, I have seen the comment indeed, but this is how I retrieved it from the original and other components. We could change all of those .format to fstrings in another refactor PR.

Comment (vlebourl) [reply custom_components/tahoma/sensor.py:106]: fine by me.

Comment (vlebourl) [reply custom_components/tahoma/sensor.py:106]: https://developers.home-assistant.io/docs/development_guidelines/#use-new-style-string-formatting

Comment (iMicknl) [reply custom_components/tahoma/sensor.py:106]:
```
self.current_value = f"{self.tahoma_device.active_states.get(CORE_RELATIVE_HUMIDITY_STATE):.2f}"
```

I am not sure if this is more readable in this case, so I will leave it for now.

**Pull request merged**